src/NServiceBus.AzureStoragePersistence/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -152,155 +155,90 @@ public void Add(TimeoutData timeout)
-                                        Time = timeout.Time,
-                                        OwningTimeoutManager = timeout.OwningTimeoutManager,
-                                        Headers = headers
-                                    };
-                var addOperation = TableOperation.Insert(timeoutData);
-                timeoutDataTable.Execute(addOperation);
-            }
-            timeout.Id = identifier;
-
-            if (timeout.SagaId != default(Guid) && !TryGetTimeoutData(timeoutDataTable, timeout.SagaId.ToString(), identifier, out timeoutDataEntity))
-            {
-                var timeoutData = new TimeoutDataEntity(timeout.SagaId.ToString(), identifier)
-                                    {
-                                        Destination = timeout.Destination.ToString(),
-                                        SagaId = timeout.SagaId,
-                                        StateAddress = identifier,
-                                        Time = timeout.Time,
-                                        OwningTimeoutManager = timeout.OwningTimeoutManager,
-                                        Headers = headers
-                                    };
-
-                var addOperation = TableOperation.Insert(timeoutData);
-                timeoutDataTable.Execute(addOperation);
-            }
+                SaveCurrentTimeoutState(timeout.State, identifier),
+                SaveTimeoutEntry(timeout, timeoutDataTable, identifier, headers),
+                SaveSagaEntry(timeout, timeoutDataTable, identifier, headers)
+            };
 
-            var timeoutDataObject = new TimeoutDataEntity(identifier, string.Empty)
-                                {
-                                    Destination = timeout.Destination.ToString(),
-                                    SagaId = timeout.SagaId,
-                                    StateAddress = identifier,
-                                    Time = timeout.Time,
-                                    OwningTimeoutManager = timeout.OwningTimeoutManager,
-                                    Headers = headers
-                                };
-            var addEntityOperation = TableOperation.Insert(timeoutDataObject);
-            timeoutDataTable.Execute(addEntityOperation);
+            await Task.WhenAll(saveActions).ConfigureAwait(false);
+            await SaveMainEntry(timeout, identifier, headers, timeoutDataTable).ConfigureAwait(false);
         }
 
         /// <summary>
         /// Peek at an existing timeout entry
         /// </summary>
         /// <param name="timeoutId">The ID of the timeout that is being requested</param>
+        /// <param name="context">The current pipeline context</param>
         /// <returns>The requested timeout entry</returns>
-        public TimeoutData Peek(string timeoutId)
+        public Task<TimeoutData> Peek(string timeoutId, ContextBag context)
```

Comment:
@dbelcham uhh wait, scratch that, seems to be removed already...

![uhh](https://media.giphy.com/media/XItRQJP0wai7m/giphy.gif)